tt_metal/Glossary.md

```diff
@@ -0,0 +1,145 @@
+- **CCL -** Collective Communication Library - The CCL provides a collection of pre-written multi-chip operations.
+- **CI -** Continuous Integration
+- **CQ -** Command Queue
+- **CNN -** Convolution Neural Network - A deep learning model suited for data processing.
+- **DMA -** Direct Memory Access
+- **DNN -** Deep Neural Network
+- **DP -** Data Parallel
+- **DRAM -** Dynamic Random Access Memory - Off-chip Memory. Provides larger, off-chip storage for the system.
+- **EDM -** ERISC Data Mover - Reusable data movement component used to send data over ethernet links. It provides ethernet link utilization and handles design concerns by implementing multi-chip collective operations.
+- **ELF -** Executable and Linkable Format - Standard file format for executable files.
+- **FDE -** Fetch-decode-execute cycle.
+- **FET -** Field Effect Transistor
+- **FPF -** Floating Point Format
+- **FVC -** Fabric Virtual Channel
+- **GBD -** GNU Debugger tool.
+- **GDDR -** Graphics Double Data Rate - Memory type used for GPUs.
+- **GH -** GitHub
+- **GPU -** Graphic Processing Unit
+- **HBM -** High Bandwidth Memory
+- **HPC -** High Performance Computing
+- **I2S - S2I -** Interleaved to Sharded, Sharded to Interleaved.
+- **ISA -** Instruction-Set Architecture - An abstract computer model that determines how the CPU is controlled by the software.
+- **ISS -** Instruction Set Simulator
+- **JVM -** Java Virtual Machine
+- **KWARG -** Keyword Argument - used client side to allow all arguments, not used in server side functions.
+- **LLK -** Low Level Kernels - Bottom layer of software stack, responsible for basic system operations.
+- **LLM -** Large Language Model
+- **LPDDR4 -** DRAM - Off-chip Memory. Provides larger, off-chip storage for the system.
+- **LUT -** Look Up Table - Mapping table of input and output values used for addressing.
+- **MLP -** Multilayer Perceptron - A modern feedforward artificial neural network.
+- **MLIR -** Multi-Level Intermediate Representation
+- **MM - MatMul - Matrix Multiplication -** Multiplication of two square or rectangular matrices.
+- **MIMD -** Multiple Instruction Multiple Data - Allows multiple processors to function simultaneously and asynchronously.
+- **NM -** Nanometers
+- **NMC -** Near Memory Computing - Computing paradigm that moves data processing closer to where the data is stored.
+- **NLP -** Natural Language Processing
+- **NoC -** Network on Chip
+- **NPU -** Neural Processing Unit
+- **NUMA -** Non-uniform Memory Access
+- **PCC -** Pearson Correlation Coefficient - A method of measuring a linear correlation between two variables.
+- **Q, K, V values -** Q, K, and V stand for different values. Q = query, K = key, V = value. In transformer architecture these values are used to help the model focus on relevant parts of the input sequence.
+- **RB -** Routing Buffer
+- **RDMA -** Remote Direct Memory Access
+- **SDK -** Software Development Kit
+- **seqL -** Sequence Length
```

Comment:
Please remove this. It's just someone's preference as an abbreviation, not TT related at all.